include/mlir/ExecutionEngine/OptUtils.h

```diff
@@ -41,9 +42,12 @@ void initializeLLVMPasses();
 
 /// Create a module transformer function for MLIR ExecutionEngine that runs
 /// LLVM IR passes corresponding to the given speed and size optimization
-/// levels (e.g. -O2 or -Os).
+/// levels (e.g. -O2 or -Os). If not null, \p targetMachine is used to
+/// initialize passes that provide target-specific information to the LLVM
+/// optimizer.
 std::function<llvm::Error(llvm::Module *)>
-makeOptimizingTransformer(unsigned optLevel, unsigned sizeLevel);
+makeOptimizingTransformer(unsigned optLevel, unsigned sizeLevel,
+                          llvm::TargetMachine *targetMachine = nullptr);
```

Comment:
I would not even have a default nullptr value here to avoid forgetting it at call-sites.